### ISSUE TITLE
Add option to grant access to public repos only

### ISSUE THREAD

**Issue by Naatan** [NirmalScaria/le-git-graph#20]
First off; thanks so much for the Firefox support! Sadly though I immediately ran into a show-stopper. It seems the extension relies on authorizing an app to have access to ALL private repos across all my organizations. I only intend to use it for certain public repos though. Any chance the app requirement could be removed for public repo's? Or make the permission project or public project specific? 

Basically, I have no desire to use this with private projects, so giving it access to them seems like a needless security risk I'd prefer avoiding.

Thanks!

**Comment by NirmalScaria** [comment]
Hey! I understand the concern about security. Pretty sure there will be many others who uninstalled the extension seeing the repo permission asked.

There is no way to make the permission project specific, but it is possible to limit the permission to public repos only. (Looking at the documentation of OAuth, it seems so. I haven't tested yet).

Im little confused how should the UI for prompting the authentication though. User will have to choose the permission level from the first page itself. Will try to figure it out soon. 😄

**Comment by NirmalScaria** [comment]
[UPDATE] This has been added. Already published for firefox. Pending review for chrome. 😄

**Comment by Naatan** [comment]
Sorry for the late response; thank you @NirmalScaria, much appreciated!
